src/taskpane/server.js

```diff
@@ -1202,13 +1314,53 @@ function protectColumn(sheet, columnNumber, rowLength, bgColor, name, hide) {
 
 }
 
+// resets ID column backgroung colors to its original - used before a GET command
+// @param: sheetName - current sheet name
 
+function resetIdColors(requiredSheetName) {
+  Excel.run(function (ctx) {
+    var rangeAddress = "A2:A5000";

```

Comment:
this range should not be hard coded for 3 reasons:
1. test step IDs are in column B
2. users may have more than 5000 rows (or far fewer)
3. it is never good to hard code things like this if you can avoid it.

Reply:
I'll need help to find a better place for this function because where it is right now, the only argument I have to pass to it is the sheet name - in order to determine what should be the IDs range we are modifying, I'll need extra information.

Reply:
I completely redraw this function - now it gets the whole range of old data and restores the background colors of each column to their original - not only the IDs. They're not hard-coded values anymore.